backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java

```diff
@@ -2140,52 +2147,55 @@ public List<ReleaseLink> getReleaseLinksOfProjectNetWorkByTrace(List<String> tra
-            releaseLink.setName(releaseById.getName());
-            releaseLink.setVersion(releaseById.getVersion());
-            releaseLink.setLongName(SW360Utils.printFullname(releaseById));
-            releaseLink.setClearingState(releaseById.getClearingState());
-            releaseLink.setComponentType(componentById.getComponentType());
-            releaseLink.setLicenseIds(releaseById.getMainLicenseIds());
-            releaseLink.setOtherLicenseIds(releaseById.getOtherLicenseIds());
-            releaseLink.setAccessible(componentDatabaseHandler.isReleaseActionAllowed(releaseById, user, RequestedAction.READ));
-            releaseLink.setNodeId(releaseById.getId() + "_" + UUID.randomUUID());
-            releaseLink.setParentNodeId(parentId);
-            releaseLink.setLayer(layer);
-            releaseLink.setProjectId(projectId);
+        for (ReleaseNode releaseNode : releaseNodes) {
+            ReleaseLink releaseLink = convertReleaseNodeToReleaseLink(releaseNode, projectId, user, parentId, layer);
             releaseLink.setIndex(index);
-            releaseLink.setReleaseMainLineState(releaseById.getMainlineState());
-            releaseLink.setAttachments(releaseById.getAttachments() != null ? Lists.newArrayList(releaseById.getAttachments()) : Collections.emptyList());
-            if (releaseById.getVendor() != null) {
-                releaseLink.setVendor(releaseById.getVendor().getFullname());
-            } else {
-                releaseLink.setVendor("");
-            }
             index++;
             releaseLinks.add(releaseLink);
         }
         return releaseLinks;
     }
 
+    protected List<ReleaseLink> convertReleaseNodesToReleaseLinksParallel(List<ReleaseNode> releaseNodes, String projectId, User user) {
+        ExecutorService executor = Executors.newFixedThreadPool(10);
+
+        final List<Callable<ReleaseLink>> callableTasksToConvertReleaseNodes = new ArrayList<>();
+        releaseNodes.forEach(releaseNode -> {
+            Callable<ReleaseLink> convertToReleaseLink = () ->
+                    convertReleaseNodeToReleaseLink(releaseNode, projectId, user, "", 0);
+            callableTasksToConvertReleaseNodes.add(convertToReleaseLink);
+        });
+
+        List<Future<ReleaseLink>> releaseLinksFuture;
+        try {
+            releaseLinksFuture = executor.invokeAll(callableTasksToConvertReleaseNodes);
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Error when convert releaseLink: " + e.getMessage());
+        }
+
+        List<ReleaseLink> releaseLinks = releaseLinksFuture.stream().map(fut -> {
+            try {
+                return fut.get();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException("Error when convert releaseLink: " + e.getMessage());
+            }
+        }).collect(Collectors.toList());
+
+        executor.shutdown();
```

Comment:
It would be good to close it inside a `finally` block

Reply:
@smrutis1, thanks for your comment, I have moved it to finally block, please help me to review it